{
 "cells": [
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2026-01-31T22:32:53.314412Z",
     "start_time": "2026-01-31T22:32:53.299408Z"
    }
   },
   "cell_type": "markdown",
   "source": "### Data Cleaning v01",
   "id": "3aae2c815781abe5"
  },
  {
   "metadata": {
    "collapsed": true
   },
   "cell_type": "code",
   "source": "import pandas as pd",
   "id": "3451581d10d7fb71",
   "outputs": [],
   "execution_count": null
  },
  {
   "metadata": {},
   "cell_type": "code",
   "source": [
    "df = pd.read_csv('data/variant_summary.txt', sep='\\t')\n",
    "df.head()"
   ],
   "id": "8fa413dac668abff",
   "outputs": [],
   "execution_count": null
  },
  {
   "metadata": {},
   "cell_type": "code",
   "source": [
    "print(df.shape)\n",
    "print(df['ClinicalSignificance'].value_counts())"
   ],
   "id": "ea1d3c4c7eba1c34",
   "outputs": [],
   "execution_count": null
  },
  {
   "metadata": {},
   "cell_type": "code",
   "source": [
    "# there are many ambiguous labels based on the ClinicalSignificance value counts\n",
    "# resolution is to exclude those from the used data\n",
    "labels_to_include = ['Pathogenic', 'Likely pathogenic', 'Benign', 'Likely benign']\n",
    "df = df[df['ClinicalSignificance'].isin(labels_to_include)]"
   ],
   "id": "66872b7d7d1aaf46",
   "outputs": [],
   "execution_count": null
  },
  {
   "metadata": {},
   "cell_type": "code",
   "source": [
    "# focusing on simple nucleotide variant for now\n",
    "df = df[df['Type'] == 'single nucleotide variant']"
   ],
   "id": "8f5a4433e9a0a64b",
   "outputs": [],
   "execution_count": null
  },
  {
   "metadata": {},
   "cell_type": "code",
   "source": [
    "print(df['ReviewStatus'].value_counts())\n",
    "\n",
    "# filter out reliable data (reviewed) data\n",
    "high_quality = [\n",
    "    'practice guideline',\n",
    "    'reviewed by expert panel',\n",
    "    'criteria provided, multiple submitters, no conflicts'\n",
    "]\n",
    "df = df[df['ReviewStatus'].isin(high_quality)]"
   ],
   "id": "a5c27570377887d",
   "outputs": [],
   "execution_count": null
  },
  {
   "metadata": {},
   "cell_type": "code",
   "source": [
    "# check for missing data in columns of interest\n",
    "important_cols = ['GeneSymbol', 'Type', 'ClinicalSignificance', 'Assembly']\n",
    "print(df[important_cols].isnull().sum())"
   ],
   "id": "aa89adcf0857bb4d",
   "outputs": [],
   "execution_count": null
  },
  {
   "metadata": {},
   "cell_type": "code",
   "source": [
    "# check genome builds\n",
    "print(df['Assembly'].value_counts())\n",
    "\n",
    "# pick one assembly to avoid duplicates\n",
    "# GRCh38 is the current reference\n",
    "df = df[df['Assembly'] == 'GRCh38']"
   ],
   "id": "4807e4ee026c83fc",
   "outputs": [],
   "execution_count": null
  },
  {
   "metadata": {},
   "cell_type": "code",
   "source": [
    "# drop duplicate variants (no use to have them in there, no additional information is gained)\n",
    "df = df.drop_duplicates(subset=['#AlleleID'], keep='first')"
   ],
   "id": "430975d90726e6",
   "outputs": [],
   "execution_count": null
  },
  {
   "metadata": {},
   "cell_type": "code",
   "source": [
    "print(df['Chromosome'].value_counts())\n",
    "# remove non-standard instances for the sake of simplicity\n",
    "# keep 1-22, X, and Y\n",
    "standard = [str(i) for i in range(1, 23)] + ['X', 'Y']\n",
    "df = df[df['Chromosome'].isin(standard)]"
   ],
   "id": "483ab2b171bfcb8d",
   "outputs": [],
   "execution_count": null
  },
  {
   "metadata": {},
   "cell_type": "code",
   "source": "df = df.reset_index(drop=True)",
   "id": "f605fc2f4742ecc9",
   "outputs": [],
   "execution_count": null
  },
  {
   "metadata": {},
   "cell_type": "code",
   "source": "df.head()",
   "id": "aee6987892384bfd",
   "outputs": [],
   "execution_count": null
  },
  {
   "metadata": {},
   "cell_type": "code",
   "source": [
    "# filter out relevant features\n",
    "features_to_keep = [\n",
    "    'Type',\n",
    "    'GeneSymbol',\n",
    "    'Chromosome',\n",
    "    'Start',\n",
    "    'Stop',\n",
    "]\n",
    "\n",
    "y = df['ClinicalSignificance'].copy()\n",
    "X = df[features_to_keep].copy()\n",
    "print('Features:')\n",
    "print(X.head())\n",
    "print('\\nTarget:')\n",
    "print(y.head())"
   ],
   "id": "a07263cf81ca8b8d",
   "outputs": [],
   "execution_count": null
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
